### DISCUSSION TITLE
How to call a cmd on bufenter after session restore?

### DISCUSSION THREAD

**Discussion by kapral18** [LazyVim/LazyVim#1738] [Q&A]
I am trying to zz on entering buffer / window

```lua
-- in lua/config/autocmds.lua

-- zz center cursor
vim.api.nvim_create_autocmd({ "BufEnter", "BufWinEnter" }, {
  pattern = { "*" },
  callback = function()
    vim.cmd.normal("zz")
  end,
})
```

But for some reason if I open the buffer with `restore session` from dashboard, this autocmd doesn't get triggered.

I tried to debug with print(), and I don't get anything logged on session restore, but it DOES get logged on any other explicit buffer open event.

Can anybody explain why this happens and how to fix this?

Thanks

**Comment by kapral18** [comment]
@folke mind if I bother you since you wrote the plugin? :)

**Comment by folke** [comment]
No idea why it works like that, but persistence uses regular Neovim sessions. I'm not doing anything weird regarding events or anything.

**Reply by kapral18** [reply]
I tried

```lua
vim.api.nvim_create_autocmd({ "SessionLoadPost" }, {
  pattern = { "*" },
  callback = function()
    print("happens")
  end,
})
```

but still no luck after loading the session from dashboard
